src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -104,6 +105,52 @@ internal void Skip()
             }
         }
 
+        internal IList<T> ReadObjectAsList<T>(IUtf8JsonStreamReaderConverter<T> streamReaderConverter)
+        {
+            if (TokenType != JsonTokenType.StartObject)
+            {
+                return Array.Empty<T>();
+
+            }
+            //We use JsonObjects for the arrays so we advance to the first property in the object which is the name/ver of the first library
+            Read();
+
+            if (TokenType == JsonTokenType.EndObject)
+            {
+                return Array.Empty<T>();
+            }
+
+            var listObjects = new List<T>();
+            do
+            {
+                listObjects.Add(streamReaderConverter.Read(ref this));
+                //At this point we're looking at the EndObject token for the object, need to advance.
+                Read();
+            }
+            while (TokenType != JsonTokenType.EndObject);
+            return listObjects;
+        }
+
+        internal void ReadArrayOfObjects<T1, T2>(IList<T2> objectList, IUtf8JsonStreamReaderConverter<T1> streamReaderConverter) where T1 : T2
+        {
+            if (objectList is null)
+            {
+                return;
+            }
+
+            if (Read() && TokenType == JsonTokenType.StartArray)
```

Comment:
if this expression returns false, I think it could hide a json schema exception. This violates the principal of failing early, so that errors are clearly understood.

For example, if there's an object `{ "something": 5, "another": "value" }`, where the `something` should be an array of objects, this code will silently ignore the error, and the `another` property would still be correctly read, assuming it's supposed to have a string value. The result would potentially be deserialized into an object with an invalid state (if `something` is not supposed to be nullablle/empty), rather than throwing an exception to report the json error.

Reply:
Updated it to check throw an error if it's called without an array.